ODLA/platforms/odla_popart/odla_compute.cc

```diff
@@ -168,26 +164,65 @@ odla_status odla_DestroyContext(odla_context ctx) {
 }
 
 odla_status odla_DestroyComputation(odla_computation comp) {
+  comp_initialized = false;
+  if (comp->session != nullptr) {
+    comp->session->getDevice().getDeviceInfo()->detach();
+    comp->session.reset();
+  }
   // g_comp.reset();
   return ODLA_SUCCESS;
 }
 
 odla_status odla_ExecuteComputation(odla_computation comp, odla_context context,
                                     odla_compute_mode mode,
                                     odla_device device) {
+  if (!comp_initialized) {
```

Comment:
it will make the first execution very slow. If we do initialization "odla_CreateContext" like before, we can avoid this problem